swig/cpp/src/Tree_Schema.hpp

```diff
@@ -126,6 +127,15 @@ class Submodule
 public:
     /** wrapper for struct [lys_submodule](@ref lys_submodule), for internal use only */
     Submodule(struct lys_submodule *submodule, S_Deleter deleter);
+    /** creates a Submodule from `module` if it is a submodule */
+    Submodule(S_Module module):
+        submodule((lys_submodule*)module->module),
+        deleter(module->deleter)
+    {
+        if (module->type() != 1) {
+            throw std::invalid_argument("Type of module must be 1 (submodule)");
```

Comment:
I think that this would be a more useful error message: `Attempted to cast a YANG module into a YANG submodule`

Reply:
done